### ISSUE TITLE
[Bug]: `lakectl ingest` for Azure doesn't allow reading ingested files 

### ISSUE THREAD

**Issue by MigQ2** [treeverse/lakeFS#5047]
### TL;DR:

`lakectl ingest` purpose is to add Data from external sources into LakeFS, but for Azure it does not work. 

## Current Behavior:

Azure storage accounts need an authentication key to read data. [According to LakeFS documentation](https://docs.lakefs.io/howto/import.html#using-the-lakectl-ingest-command) when doing a `lakectl ingest` I should add the storage account key as a local environment variable. That makes the `ingest` command authenticate and seems to ingest the data successfully.
        
However, when trying to read that data using LakeFS `s3a://` afterwards I get an error, because the key of the storage account hasn't been stored anywhere and I have no way of passing it to LakeFS. 

### Why is this important? 

Currently I can do `lakectl import` to load data from the same storage account that LakeFS uses as underlying storage, but there is no way to load data from other storage accounts, **For organizations with multiple Azure storage accounts this becomes a blocker for LakeFS adoption**

## Steps to Reproduce:
1. Do a `lakectl ingest` from a different Azure storage account than the one LakeFS uses as underlying storage
2. Try to read the data, e.g. with a `spark.read.csv("s3a://<some-lakefs-path>")` 

## Additional comments

- The `lakefs ingest` [documentation](https://docs.lakefs.io/howto/import.html#using-the-lakectl-ingest-command) is very misleading: in the prerequisites it states that the user and LakeFS installation should have permissions to list and read the objects being ingested, but there's no way I can give LakeFS permission to an Azure storage account
- I guess the easiest way of solving this is allowing LakeFS to store multiple keys for Azure Storage accounts and automatically use the corresponsing key to authenticate whenever I ingest or read data from any of them

### lakeFS Version

0.86.0

### Deplyoment

Kubernetes/Helm in Azure

### Contact Details

Iddo and Amit have my email. You can also reach me in LakeFS community Slack

**Comment by itaiad200** [comment]
Thanks @MigQ2 for the detailed explanation. I think you nailed the root cause: lakeFS must have access to the ingested/imported objects in order to be able to read them. 
Indeed there's currently no way of passing additional creds to lakeFS. I'm not very familiar with Azure storage accounts auth, but isn't it possible to have a user/serviceaccount that can access more than a single storage account? Then, instead of giving more creds to lakeFS, we can just add more permissions to the user/serviceaccount lakeFS already uses.
Using multiple creds in lakeFS can be tricky - almost every access to the storage layer should be retried with different set of creds.

**Comment by MigQ2** [comment]
Thanks for your reply @itaiad200. 

I see the [lakeFS Azure authentication configuration](https://docs.lakefs.io/reference/configuration.html#reference) seems to support `blockstore.azure.auth_method='msi'`. 
However I don't see clearly how that works because all the examples use an account_name and account_key to authenticate.

Do you know how can I configure LakeFS to authenticate using `msi` and a specific user or service principal? Maybe that way we can fix it.

**Comment by itaiad200** [comment]
@MigQ2 We were just updating our Azure SDK (available in [v0.90.0](https://github.com/treeverse/lakeFS/releases/v0.90.0)), and we have a [documentation gap](#4931) that we need to fill. I haven't tested it yet, but from the [Azure docs](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication?tabs=bash#-option-2-use-a-managed-identity) it seems as easy as setting the `AZURE_CLIENT_ID` env var. I can test it later this week and report here. If you test it yourself, make sure to use the latest version, configure `msi` in the lakeFS config (`blockstore.azure.auth_method`) and use that env var.

**Comment by MigQ2** [comment]
That sounds good, thanks for the info! I'm not sure I'll have the time this week to test that, but let's keep in touch and post here anything we find out.

**Comment by MigQ2** [comment]
Thank you @itaiad200  for the details. I have a couple questions/comments on your last update: 

1. When you say "import" you really mean "ingest", right?
2. I was going to test `msi` this week. Should I give it a try or you know for a fact that it's broken and needs a fix from your side?

**Comment by itaiad200** [comment]
1. `Import` and `Ingest` are 2 ways of pointing to an external objects from a lakeFS repository. I'm testing both, import is easier to start with as it requires only server-side credentials. See [more](https://docs.lakefs.io/howto/import.html#import-data-into-lakefs) on the differences between the two.
2. I tested it now, it works with `v0.90.1`. I had to assign the vm-identity with `Contributor`, `Storage Blob Data Contributor` and `Storage Queue Data Contributor` roles for the storage account of the repository storage namespace. For the storage account of the imported objects I used the same roles, but I guess you can replace `Contributor` with `Reader` where it's applicable.

**Comment by MigQ2** [comment]
Thanks @itaiad200.

1. By reading the documentation I had the impression that `import` was intended to be used to import data from the storage account that is used as lakeFS underlying storage. And `ingest` from different storage accounts. If this is not the case the documentation doesn't make it so clear to me on when should I use each of them
2. Sounds great! If it works good for you I will test on my side as well
